Sources/XMTPiOS/CodecRegistry.swift

```diff
@@ -8,7 +8,10 @@
 import Foundation
 
 struct CodecRegistry {
-	var codecs: [String: any ContentCodec] = [TextCodec().id: TextCodec()]
+	var codecs: [String: any ContentCodec] = [
+		TextCodec().id: TextCodec(),
+		GroupMembershipChangedCodec().id: GroupMembershipChangedCodec()
```

Comment:
@richardhuaaa and I discussed this yesterday and we actually think it doesn't make sense to auto register `GroupMembershipChangedCodec` since it would require rendering code to be supported on the client side and would break if they don't support that.

Reply:
That makes sense to me.

We should discuss if we want fallback text here. I'm leaning towards no, since it's a little weird to just have auto-generated messages that you didn't send. If these aren't rendered distinctly from normal conversation messages I think the experience is going to be weird.

Reply:
Yea, when I do the rename of Conversation -> DirectMessage then I'll introduce a flag to officially enable group chat and we can conditionally register this codec then.

For now, the only time a user would ever see this type of content is if they explicitly load groups via `conversations.groups()`, the old `list` method is unchanged here.

Reply:
> we can conditionally register this codec then

So it might make sense to actually let the app developer register this codec, if and only if they add code to handle the codec - even if they enable group chats their app might crash otherwise.

Reply:
Ok I'm sold, making it opt-in.

Reply:
Thats a valid point but I still don't think we should auto opt-in I think instead maybe we need a fallback text.